### PR TITLE
fix(android, crash): call sign in with Apple service on UI thread

### LINKED ISSUE #195: Android crash - Method addObserver must be called on the main thread
Hello, I always receive the error when calling the method `signIn` on Android

**Code example:**
```ts
            if (!appleAuthAndroid.isSupported) {
                throw new Error('Unsupported version of Android');
            }
            try {
                const rawNonce = generateStringUID();
                const state = generateStringUID();
                appleAuthAndroid.configure({
                    clientId: 'http://xxx',
                    redirectUri: 'https://xxx/callback',
                    responseType: appleAuthAndroid.ResponseType.ALL,
                    scope: appleAuthAndroid.Scope.ALL,
                    nonce: rawNonce,
                    state,
                });

                const appleAuthRequestResponse = await appleAuthAndroid.signIn();
                if (appleAuthRequestResponse.state !== state) {
                    throw new Error('Cannot sign in with Apple');
                }
                return {
                    idToken: appleAuthRequestResponse.id_token!,
                    authorizationCode: appleAuthRequestResponse.code,
                    nonce: rawNonce,
                };
            } catch (error) {
                switch (error.message) {
                    case appleAuthAndroid.Error.NOT_CONFIGURED:
                    case appleAuthAndroid.Error.SIGNIN_FAILED:
                        throw new Error('Cannot sign in with Apple');
                    case appleAuthAndroid.Error.SIGNIN_CANCELLED:
                        throw CanceledError;
                }
                throw error;
            }
```

**Error:**
```java.lang.IllegalStateException: Method addObserver must be called on the main thread
8:33 2021-03-12 07:27:07.969 30003-30196/com.tullyapp.test E/unknown:ReactNative: Exception in native call
    java.lang.IllegalStateException: Method addObserver must be called on the main thread
        at androidx.lifecycle.LifecycleRegistry.enforceMainThreadIfNeeded(LifecycleRegistry.java:317)
        at androidx.lifecycle.LifecycleRegistry.addObserver(LifecycleRegistry.java:172)
        at androidx.fragment.app.Fragment.initLifecycle(Fragment.java:496)
        at androidx.fragment.app.Fragment.<init>(Fragment.java:476)
        at androidx.fragment.app.DialogFragment.<init>(DialogFragment.java:138)
        at com.RNAppleAuthentication.webview.SignInWebViewDialogFragment.<init>(SignInWebViewDialogFragment.kt:18)
        at com.RNAppleAuthentication.webview.SignInWebViewDialogFragment$Companion.newInstance(SignInWebViewDialogFragment.kt:24)
        at com.RNAppleAuthentication.SignInWithAppleService.show(SignInWithAppleService.kt:80)
        at com.RNAppleAuthentication.AppleAuthenticationAndroidModule.signIn(AppleAuthenticationAndroidModule.java:232)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.facebook.react.bridge.JavaMethodWrapper.invoke(JavaMethodWrapper.java:372)
        at com.facebook.react.bridge.JavaModuleWrapper.invoke(JavaModuleWrapper.java:151)
        at com.facebook.react.bridge.queue.NativeRunnable.run(Native Method)
        at android.os.Handler.handleCallback(Handler.java:883)
        at android.os.Handler.dispatchMessage(Handler.java:100)
        at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage(MessageQueueThreadHandler.java:27)
        at android.os.Looper.loop(Looper.java:214)
        at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run(MessageQueueThreadImpl.java:226)
        at java.lang.Thread.run(Thread.java:919)
```

**[Updated] Possible solution in bridge:**
```java
getCurrentActivity().runOnUiThread(
        new Runnable() {
            @Override
            public void run() {
                service.show();
            }
        }
);
```

**Env**:
* react-native: 0.62.2
* react-native-apple-authentication: 2.1.0

### PR DESCRIPTION
**Pull request by yuri-lomashko-itechart**
Fixes #195

### CODE REVIEW

File: android/src/main/java/com/RNAppleAuthentication/AppleAuthenticationAndroidModule.java [hunk 1]
```diff
@@ -229,6 +229,16 @@ public void onSignInWithAppleCancel() {
                 callback
         );
 
-        service.show();
+        Activity activity = getCurrentActivity();
+        if (activity == null) {
+            promise.reject(new RuntimeException("Activity is not found"));
```
Comment (mikehardy) [inline android/src/main/java/com/RNAppleAuthentication/AppleAuthenticationAndroidModule.java:234]: thanks for checking that :-), working across the JS<->native bridge means that every once in a while the current activity is null by the time you try to use it, and it causes NullPointerExceptions :-)

**Review by mikehardy — APPROVED** [review]
Nice one! Thanks for the fix

File: android/src/main/java/com/RNAppleAuthentication/AppleAuthenticationAndroidModule.java (hunk 1 above)
Comment (yuri-lomashko-itechart) [reply android/src/main/java/com/RNAppleAuthentication/AppleAuthenticationAndroidModule.java:234]: Yeah, no problem :)

**Comment by mikehardy** [comment]
2.1.1 is out with this :rocket: 

https://github.com/invertase/react-native-apple-authentication/releases/tag/v2.1.1

**Pull request merged**